src/providers/oracle/qgsoracleconn.cpp

```diff
@@ -192,6 +192,15 @@ void QgsOracleConn::disconnect()
   deleteLater();
 }
 
+void QgsOracleConn::reconnect()
+{
+  if ( mDatabase.isOpen() )
+  {
+    mDatabase.close();
+    mDatabase.open();
```

Comment:
@jef-n Do QGIS re-open the connection only when the connection was open ?

Do I use credentials like line 99, to re-open the connection ? https://github.com/qgis/QGIS/blob/520ef21a5d94de47f948afe3c6973ab3c580b1a0/src/providers/oracle/qgsoracleconn.cpp#L97

Reply:
@jef-n what is your point of view about the way to re-open connection ?

Reply:
Hi @jef-n are agree with this point ?

Reply:
Does it work?

Reply:
it works...